src/clients/SpokePoolClient.ts

```diff
@@ -835,6 +836,15 @@ export class SpokePoolClient extends BaseAbstractClient {
           ? { ...(spreadEventWithBlockNumber(event) as V3FillWithBlock), destinationChainId: this.chainId }
           : { ...(spreadEventWithBlockNumber(event) as V2FillWithBlock) };
 
+        if (isV3Fill(fill) && Array.isArray(fill.relayExecutionInfo)) {
```

Comment:
@pxrl I think if you merge in https://github.com/across-protocol/sdk-v2/pull/596 we won't need this isV3Fill check

Reply:
Yeah agree. I'm fine for this to go in as-is though - I can update my PR to subsequently remove it.